addons/stock/views/stock_inventory_views.xml

```diff
@@ -13,7 +13,67 @@
                 <field name="outdated" invisible="1"/>
-                <field name="product_id" options="{'no_create': True}" width="1.6"
+                <field name="product_id" width="1.6"
+                       context="{'default_type': 'product'}"
+                       readonly="context.get('default_product_id', False)"
+                       attrs="{'readonly': ['|',
+                            ('is_editable', '=', False),
+                            ('state', '!=', 'confirm')]}"/>
+                <field name="location_id" groups="stock.group_stock_multi_locations"
+                       options="{'no_create': True}"
+                       invisible="context.get('readonly_location_id', False)"
+                       readonly="context.get('readonly_location_id', False)"
+                       attrs="{'readonly': ['|',
+                            ('is_editable', '=', False),
+                            ('state', '!=', 'confirm')]}"/>
+                <field name="inventory_date" optional="hide" width="0.8"/>
+                <field name="prod_lot_id" groups="stock.group_production_lot" width="0.8"
+                       attrs="{'readonly': ['|', '|',
+                            ('product_tracking', '=', 'none'),
+                            ('is_editable', '=', False),
+                            ('state', '!=', 'confirm')]}"
+                       context="{'default_product_id': product_id, 'default_company_id': company_id}" optional="show"/>
+                <field name="package_id" groups="stock.group_tracking_lot" width="0.8"
+                       attrs="{'readonly': ['|',
+                            ('is_editable', '=', False),
+                            ('state', '!=', 'confirm')]}"
+                       string="Package" optional="show"/>
+                <field name="partner_id" groups="stock.group_tracking_owner"
+                       attrs="{'readonly': ['|',
+                            ('is_editable', '=', False),
+                            ('state', '!=', 'confirm')]}"/>
+                <field name="theoretical_qty" string="On Hand" width="0.5"
+                       force_save="1" readonly="1" optional="show"/>
+                <button name="action_refresh_quantity" title="Refresh quantity"
+                        icon="fa-refresh" type="object"
+                        attrs="{'invisible': [('outdated', '=', False)]}"/>
+                <field name="product_qty" width="0.5" string="Counted"/>
+                <field name="difference_qty" optional="show" width="0.5"/>
+                <field name="product_uom_id" string="UoM" groups="uom.group_uom"
+                       width="0.3" force_save="1"/>
+                <field name="inventory_id" invisible="1"/>
+                <field name="state" invisible="1"/>
+            </tree>
+        </field>
+    </record>
+
+    <!-- Same as stock_inventory_line_tree except that new products cannot be created in product_id field -->
```

Comment:
You should use an inherit with a <field name="mode">primary</field>
It will copy the view an add your change.

Reply:
Done. Didn't know about inherit with primary, thanks!